preditor/gui/console.py

```diff
@@ -478,6 +538,17 @@ def keyPressEvent(self, event):
 
         completer = self.completer()
 
+        # Define prefix so we can determine if the exact prefix is in
+        # completions and highlight it. We must manually add the currently typed
+        # character, or remove it if backspace or delete has just been pressed.
+        key = event.text()
+        _, prefix = completer.currentObject(scope=__main__.__dict__)
+        isBackspaceOrDel = key in ("", "")
```

Comment:
I would prefer to avoid encoding raw bytes into text like this, especially now that I see it in github. I wonder if this would cause encoding issues later on. I think this should accomplish the same thing but be easier to read.

```py
isBackspaceOrDel = event.key() in (Qt.Key_Backspace, Qt.Key_Delete)
```